app/reducers/reducer_pins.js

```diff
@@ -22,8 +23,18 @@ export default function(state = initialState, action) {
       return Object.assign({}, state,
         newPin
       );
+    case DELETE_PIN:
+      let id = action.payload.id
+      let deletedPinState = Object.assign({}, state);
+      deletedPinState[id] = Object.assign({}, state[id]);
+      delete deletedPinState[id];      
+
+      return Object.assign({},
```

Comment:
I think it would be sufficient to simple `delete id` from the `state` copy, then return it.
```JS
case DELETE_PIN:
  const id = action.payload.id
  let newState = Object.assign({}, state);
  delete newState[id];
  return newState;
}
```